### DISCUSSION TITLE
Filter on multiple columns

### DISCUSSION THREAD

**Discussion by Eralmidia** [IgniteUI/igniteui-angular#13240] [Q&A] [answered]
Hi, guys

I was just wondering if the following is possible to achieve:

Lets say you have a grid with the following data set: 
[{a: "a1", b: undefined}, {a: "a2", b: undefined}, {a: "a2", b: "b1"}, {a: "a2", b: "b2"}]

To save screen real estate, you do not want to display a column for both a and b. Instead, you want a single column like this:
![image](https://github.com/IgniteUI/igniteui-angular/assets/12936286/6dd51ba4-5931-41ec-a737-5878462582b1)
, where the b values are just indented sub values of a.

Because of templating, this is easy to achive (just use row.data in the template to write the value of b into the column for a). However, when enabling the excel style filter for column a, the search will now only search on values for a, not b. Is there an easy way to make the filtering search for both properties in the same column?

**Comment by ddaribo** [comment, marked as answer]
Hi @Eralmidia,

The first approach to consider for this scenario would be the [Formatted Values Filtering Strategy](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/excel-style-filtering#formatted-values-filtering-strategy) feature of the Excel Style Filtering.

It allows to define a custom formatter function that is to be used for the values in the ESF filtering list. In this way you could return a string concatenating both values, as used in the cell template. However, since this function only has the bound column’s value as a parameter, some kind of mapping needs to be done to extract the other value, for example:

```
  public formatStringColumn(value: number) {
    let supplierId = '';
    const matchingRec = DATA.find((item) => item.ProductName === value);
    if (matchingRec) {
      supplierId = matchingRec.SupplierID;
    }
    return value + ' ' + supplierId;
  }
```

Of course, the logic will depend on your specific application scenario.

Alternatively, if such a mapping does not make sense for your case, another suggestion might be creating a new field for each data record whose values would concatenate the corresponding values before binding the data to the grid and simply binding that as a column on its own.

[Here](https://stackblitz.com/edit/igx-grid-esf-formatted-value-fitlering-str?file=src%2Fapp%2Fgrid%2Fgrid-formatted-filtering-strategy%2Fgrid-formatted-filtering-strategy.component.ts) is also a minimal sample on the Formatted Values approach.

If you need any further assistance on the matter, please, let me know.

**Comment by Eralmidia** [comment]
Yeah, that should work, thank you. It would be great if the formatter could take in the row data as a second argument tho, to avoid finding the correct element in the data array.
